metrics/resource_view.go

```diff
@@ -315,21 +315,49 @@ func resourceToKey(r *resource.Resource) string {
 	if r == nil {
 		return ""
 	}
+
+	// If there are no labels, we have just the Type.
+	if len(r.Labels) == 0 {
+		return r.Type
+	}
+
 	var s strings.Builder
+	writeKV := func(key, value string) {
+		// We use byte values 1 and 2 to avoid colliding with valid resource labels
+		// and to make unpacking easy
+		s.WriteByte('\x01')
+		s.WriteString(key)
+		s.WriteByte('\x02')
+		s.WriteString(value)
+	}
+
+	// If there's only one label, we can skip building and sorting a slice of keys.
+	if len(r.Labels) == 1 {
+		l := len(r.Type)
+		var key string
+		for k, v := range r.Labels {
+			key = k
+			l += len(k) + len(v) + 2
+		}
+		s.Grow(l)
+
+		s.WriteString(r.Type)
+		writeKV(key, r.Labels[key])
```

Comment:
bind `v` to `val` to avoid lookup?
Or, which will even avoid creating both stack variables just put everything in the loop.
We know it will execute only once. Even `l` will be redundant (s.Grow(len(type)+len(k)+len(v)+2))

Reply:
Good one, will do.